projects/packages/connection/src/class-nonce-handler.php

```diff
@@ -176,16 +176,26 @@ public function delete( $limit = 10, $cutoff_timestamp = null ) {
 			)
 		);
 
-		if ( ! is_array( $ids ) || ! count( $ids ) ) {
-			// There's either nothing to remove, or there's an error and we can't proceed.
+		if ( ! is_array( $ids ) ) {
+			// There's an error and we can't proceed.
+			return false;
+		}
+
+		$ids = array_filter( $ids );
+
+		if ( ! count( $ids ) ) {
+			// There's nothing to remove.
 			return false;
 		}
 
 		$ids_fill = implode( ', ', array_fill( 0, count( $ids ), '%d' ) );
 
+		$args   = $ids;
+		$args[] = 'jetpack_nonce_%';
+
 		// The Code Sniffer is unable to understand what's going on...
-		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare
-		return $wpdb->query( $wpdb->prepare( "DELETE FROM `{$wpdb->options}` WHERE `option_id` IN ( {$ids_fill} )", $ids ) );
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPress.DB.PreparedSQLPlaceholders.ReplacementsWrongNumber
+		return $wpdb->query( $wpdb->prepare( "DELETE FROM `{$wpdb->options}` WHERE `option_id` IN ( {$ids_fill} ) AND option_name LIKE %s", $args ) );
```

Comment:
Whoopsie, the final statement doesn't look right to me. It seems that the prepare statement treats the `%` in the `jetpack_nonce_%` string as a placeholder, can that be possible? I'm looking at the prepared statement and it ends with:
```
AND option_name LIKE 'jetpack_nonce_{5dfbac26bc23ead956d9c58482b7a13b1d18348dcbef58d759bed822a09d3b36}'
```

Reply:
No, wait, that's impossible, must be something wrong in my setup - still looking.

Reply:
Apparently that’s intended.
https://developer.wordpress.org/reference/classes/wpdb/placeholder_escape/

```sql
SELECT * FROM `wp_options` WHERE option_name LIKE 'jetpack_nonce_{dac3e19c42998498a8f2f1d5a0d411d905e7982ded06298d594994a7e12c9e8f}'
```

A query like this finds the `jetpack_nonce_something_something` entries.